src/session/session_api.c

```diff
@@ -1475,6 +1476,26 @@ __wt_session_range_truncate(
               session, EINVAL, "the start cursor position is after the stop cursor position");
     }
 
+    /*
+     * Utilize the cursor lower bound as the temporary buffer to storing the original start and stop
+     * key. We use the original keys for the write-ahead log.
+     */
+    if (!local_start && start != NULL) {
+        if (WT_CURSOR_BOUNDS_SET(start))
+            WT_ERR_MSG(
+              session, EINVAL, "cursor with bounds is not compatible with truncate operation.");
```

Comment:
can we make the error a bit more clear: "a cursor with bounds set cannot perform truncate."